endoreg_db/models/metadata/sensitive_meta_logic.py

```
@@ -205,13 +292,12 @@ def perform_save_logic(instance: "SensitiveMeta") -> "Examiner":
-         # Attempt to guess if names are available
-         first_name = instance.patient_first_name or DEFAULT_UNKNOWN_NAME
-         gender = guess_name_gender(first_name)
-         if not gender:
-             raise ValueError("Patient gender could not be determined and must be set before saving.")
-         instance.patient_gender = gender
-
+        # Attempt to guess if names are available
+        first_name = instance.patient_first_name or DEFAULT_UNKNOWN_NAME
+        gender = guess_name_gender(first_name)
+        if not gender:
+            raise ValueError("Patient gender could not be determined and must be set before saving.")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)